docs/windows-design.md

```diff
@@ -0,0 +1,261 @@
+and retrieves the local Subnet CIDR from the Node spec. 
+
+After the HNS Network is created, OVS extension should be enabled at once on the Hyper-V Switch.
+
+### Container network configuration
+
+[**host-local**](https://github.com/containernetworking/plugins/tree/master/plugins/ipam/host-local)
+plugin is used to provide IPAM for containers, and the address is allocated from the subnet CIDR
+configured on the HNS Network. 
+
+Windows HNS Endpoint is leveraged as the vNIC for each container. A single HNS Endpoint with the
+IP allocated by the IPAM plugin is created for each Pod. The HNS Endpoint should be attached to all
+containers in the same Pod to ensure that the network configuration can be correctly accessed (this
+operation is to make sure the DNS configuration is readable from all containers).
+ 
+One OVS internal port with the same name as the HNS Endpoint is also needed, in order to handle
+container traffic with OpenFlow rules. OpenFlow entries are installed to implement Pod-to-Pod and
+Pod-to-external connectivity. Service traffic is handled by kube-proxy, and a specific
+OpenFlow rule is installed to output these packets to the gateway port.
+
+CNIAdd request might be called multiple times for a given Pod. This is because kubelet on Windows
+assumes CNIAdd is an idempotent event, and it uses this event to query the Pod networking status.
+Antrea needs to identify the container type (sandbox or workload) from the CNIAdd request:
+* we create the HNS Endpoint only when the request is for the sandbox container
+* we attach the HNS Endpoint no matter whether it is a sandbox container or a workload container.
+
+### Gateway port configuration
+
+The gateway port is created during the Antrea Agent initialization phase, and the address of the interface
+should be the first IP in the subnet. The port is an OVS internal port and its default name is "gw0".
+
+The gateway port is used to help implement L3 connectivity for the containers, including Pod-to-external,
+Pod-to-Service, and Node-to-Pod. For Pod-to-external, and Pod-to-Service cases, OpenFlow entries are
+installed in order to output these packets to the host on the gateway port. To ensure the packet is forwarded
+correctly on the host, the IP-Forwarding feature should be enabled on the network adapter of the
+gateway port.
+
+A routing entry for traffic from the Node to the local Pod subnet is needed on the Windows host to ensure 
+that the packet can enter the OVS pipeline on the gateway port. This routing entry is added when the "gw0"
+is created.
+
+Every time a new Node joins the cluster, a host routing entry on the gateway port is required, and the
+remote subnet CIDR should be routed with the remote gateway address as the nexthop.
+
+### Tunnel port configuration
+
+Tunnel port configuration should be similar to Antrea on Linux:
+* tunnel port is created after OVS bridge is added; 
```

Comment:
s/tunnel port is created after OVS bridge is added;/the tunnel port is added after the OVS bridge is created;